resolwe_bio/processes/support_processors/bam_conversion.py

```diff
@@ -99,41 +111,53 @@ class Input:
             label="BEDPE Normalization factor",
             description="The BEDPE file describes disjoint genome features, "
             "such as structural variations or paired-end sequence alignments. "
-            "It is used to estimate the scale factor.",
+            "It is used to estimate the scale factor [--scaleFactor].",
+            required=False,
         )
         scale = FloatField(
             label="Scale for the normalization factor",
-            description="Magnitude of the scale factor. "
-            "The scaling factor is calculated by dividing the scale "
-            "with the number of features in BEDPE "
-            "(scale/(number of features)).",
+            description="Magnitude of the scale factor. The scaling factor "
+            "[--scaleFactor] is calculated by dividing the scale with the "
+            "number of features in BEDPE (scale/(number of features)).",
+            disabled="!bedpe",
             default=10000,
         )
+        bin_size = IntegerField(
+            label="Bin size[--binSize]",
+            description="Size of the bins (in bp) for the output bigWig file. "
+            "A smaller bin size value will result in a higher resolution of "
+            "the coverage track but also in a larger file size.",
+            default=50,
+        )
 
     class Output:
         """Output fields."""
 
-        bigwig = FileField(label="bigwig file")
+        bigwig = FileField(label="Coverage file (bigWig)")
         species = StringField(label="Species")
         build = StringField(label="Build")
 
     def run(self, inputs, outputs):
         """Run the analysis."""
-        path = inputs.alignment.output.bam.path
-        basename = os.path.basename(path)
+        bam_path = Path(inputs.alignment.output.bam.path)
+        basename = bam_path.name
         assert basename.endswith(".bam")
         name = basename[:-4]
-        out_file = f"{name}.SInorm.bigwig"
-        out_index = f"{name}.bai"
 
-        with open(inputs.bedpe.output.bedpe.path) as f:
-            spike_count = f.readlines()
-            spike_count = len(spike_count)
-        scale_factor = inputs.scale / spike_count
+        if inputs.bedpe:
+            with open(inputs.bedpe.output.bedpe.path, "rb") as f:
+                spike_count = sum(1 for _ in f)
+            scale_factor = inputs.scale / spike_count
```

Comment:
This could fail if `bedpe` is an empty file.

Reply:
Good point, thank you.